MetaMorpheus/EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -606,6 +607,29 @@ private static float GetMobilityZScore(SpectralMatch psm, IBioPolymerWithSetMods
             return (float)mobilityZScore;
         }
 
+        private static float GetFraggerHyperscore(SpectralMatch psm, IBioPolymerWithSetMods selectedPeptide)
+        {
+            var peptideFragmentIons = psm.BioPolymersWithSetModsToMatchingFragments[selectedPeptide];
+            var nIons = peptideFragmentIons.Where(f => f.NeutralTheoreticalProduct.Terminus == FragmentationTerminus.N).ToList();
+            var cIons = peptideFragmentIons.Where(f => f.NeutralTheoreticalProduct.Terminus == FragmentationTerminus.C).ToList();
+            ulong nIonFactorial = GetFactorial((ulong)nIons.Count);
+            nIonFactorial = nIonFactorial < 0 ? ulong.MaxValue : nIonFactorial; // this is a hack to prevent overflow
+            ulong cIonFactorial = GetFactorial((ulong)cIons.Count);
+            cIonFactorial = cIonFactorial < 0 ? ulong.MaxValue : cIonFactorial; // this is a hack to prevent overflow
```

Comment:
Ulong will overflow after 20 fragment ions. This check will not prevent overflow due to it being unsigned. We should discuss what max value is required. BigInteger does not appear to have a practical limit to its value. 
![image](https://github.com/smith-chem-wisc/MetaMorpheus/assets/96196865/12ba896b-59f2-4bbd-a22e-d01ec0a9fee0)